pkg/operation/botanist/component/vpnshoot/vpnshoot.go

```diff
@@ -684,6 +697,46 @@ func (v *vpnShoot) getVolumes(secretCA, secret, secretTLSAuth, secretDH *corev1.
 				},
 			},
 		})
+	} else {
+		hostPathCharDev := corev1.HostPathCharDev
+		volumes = append(volumes, corev1.Volume{
+			Name: volumeNameDevNetTun,
+			VolumeSource: corev1.VolumeSource{
+				HostPath: &corev1.HostPathVolumeSource{
+					Path: volumeMountPathDevNetTun,
+					Type: &hostPathCharDev,
+				},
```

Comment:
The PodSecurityPolicy has to be adapted to allow `hostPath` volumes. I think this should be also the reason why the e2e tests with `allowPrivilegedContainers=false` fails currently.

Reply:
Thanks for the hint. I was about to look into the failing test, but now it seems obvious.
I updated the pod security policy accordingly in the latest commit.